eml/components/login/LoginForm.js

```diff
@@ -22,141 +28,195 @@ import AsyncStorage from '@react-native-async-storage/async-storage';
  */
 export default function LoginForm() {
 
+    /* 
+        const [email, setEmail] = useState('');
+        const [password, setPassword] = useState('');
+        const [modalVisible, setModalVisible] = useState(false);
+        const [passwordAlert, setPasswordAlert] = useState('');
+        const [emailAlert, setEmailAlert] = useState('');
+    */
+
 	const navigation = useNavigation();
-	const [email, setEmail] = useState('');
-	const [password, setPassword] = useState('');
-	const [modalVisible, setModalVisible] = useState(false);
-	const [passwordAlert, setPasswordAlert] = useState('');
-	const [emailAlert, setEmailAlert] = useState('');
-	// State variable to track password visibility
-	const [showPassword, setShowPassword] = useState(false);
+
+    // state variable to track if user successfully submitted phone number
+    const [submittedPhoneNumber, setSubmittedPhoneNumber] = useState(false);
+
+    const [phoneNumber, setPhoneNumber] = useState('');
+	const [error, setError] = useState('');
+	const [token, setToken] = useState('');
+
+
+    // allow + symbol at start, then digits only (5 - 25 minmax length)
+    const validatePhoneNumber = (phone) => /^\+?[\d\s]{5,25}$/.test(phone);
+
+    const handlePhoneChange = (inputPhone) => {
+        setPhoneNumber(inputPhone);
+
+        if (inputPhone.length > 0) {
+            validatePhoneNumber(inputPhone) ? setError('') : setError('NÚMERO DE TELÉFONO INVÁLIDO');
+        } else {
+            setError('');
+        }
+    };
+
+    const submitPhoneNumber = (phoneNumber) => {
+
+        const res = { success: true }; //TODO: call backend and get a real res;
+
+        if(res.success){
+            setSubmittedPhoneNumber(true);
+            setError('');
+        } else {
+            setSubmittedPhoneNumber(false);
+            setError(res?.error);
+        }
+
+    }
+
```

Comment:
```suggestion
    const handlePhoneSubmit = () => {
    
        if(phoneSubmitted) {
          void login(token);
          return;
        }

        const res = { success: true }; //TODO: call backend and get a real res;

        if(res.success){
            setPhoneSubmitted(true);
            setError('');
        } else {
            setPhoneSubmitted(false);
            setError(res?.error);
        }

    }

```

Have not tested. Cast login promise to void.

No need to pass a variable which is store in state.